libipl/p10/ipl0.C

```diff
@@ -377,19 +385,84 @@ static int ipl_alignment_check(void)
 static int ipl_set_ref_clock(void)
 {
 	struct pdbg_target *proc;
+    struct pdbg_target *clock_target;
+    enum pdbg_target_status status;
 	int rc = 0;
+    uint8_t clock_count = 0;
 
 	if (ipl_type() == IPL_TYPE_MPIPL)
 		return -1;
 
 	ipl_log(IPL_INFO, "Istep: set_ref_clock: started\n");
 
+	ipl_log(IPL_INFO, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+        uint8_t data;
+
+        status = pdbg_target_probe(clock_target);
+        assert(status == PDBG_TARGET_ENABLED);
+
+        data = OSC_RESET_CMD;
+        rc = i2c_write(clock_target, 0, OSC_CTL_OFFSET, OSC_RESET_CMD_LEN, &data);
+        if(rc){
+            ipl_log(IPL_ERROR, "soft reset command is failed for clock %d, with rc = %d\n",
+                                pdbg_target_index(clock_target), rc);
+
+		    ipl_error_callback(IPL_ERR_CLK_FAILED);
+
+            return 1;
+        }
+        else{
+            ipl_log(IPL_INFO, "soft reset command is successfull for clock %d\n",
+                                pdbg_target_index(clock_target));
+        }
+
+        sleep(1);
+        rc = i2c_read(clock_target, 0, OSC_STAT_OFFSET, OSC_STAT_REG_LEN, &data);
+        if(rc){
+            ipl_log(IPL_ERROR, "status register read is failed for clock %d, with rc = %d\n",
+                                pdbg_target_index(clock_target), rc);
+
+		    ipl_error_callback(IPL_ERR_CLK_FAILED);
+            return 1;
+        }
+        else{
+            ipl_log(IPL_INFO, "status register value for clock %d is 0x%2X\n",
+                                pdbg_target_index(clock_target), data);
+            if(data & OSC_STAT_ERR_MASK){
```

Comment:
JS: Any specific bit will be set in the `data`? Looks yes, since you are using some mask. If so, please add a comment on that.